pkg/controller/prpqr_reconciler/prpqr_reconciler.go

```diff
@@ -345,10 +407,32 @@ func jobNameHash(name string) string {
 	return hex.EncodeToString(hasher.Sum(nil))
 }
 
-func generateProwjob(rc injectingResolverClient, defaulter periodicDefaulter, baseCiop *api.Metadata, prpqrName, prpqrNamespace string, spec *v1.ReleaseJobSpec, pr *v1.PullRequestUnderTest) (*prowv1.ProwJob, error) {
+func resolveCiopConfig(rc injectingResolverClient, baseCiop *api.Metadata, spec v1.ReleaseJobSpec) (*api.ReleaseBuildConfiguration, error) {
+	inject := &api.MetadataWithTest{
+		Metadata: api.Metadata{
+			Org:     spec.CIOperatorConfig.Org,
+			Repo:    spec.CIOperatorConfig.Repo,
+			Branch:  spec.CIOperatorConfig.Branch,
+			Variant: spec.CIOperatorConfig.Variant,
+		},
+		Test: spec.Test,
+	}
+
+	ciopConfig, err := rc.ConfigWithTest(baseCiop, inject)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get config from resolver: %w", err)
+	}
+
+	return ciopConfig, nil
+}
+
+type options struct {
+	labels map[string]string
+}
+
+func generateProwjob(ciopConfig *api.ReleaseBuildConfiguration, defaulter periodicDefaulter, baseCiop *api.Metadata, prpqrName, prpqrNamespace string, spec *v1.ReleaseJobSpec, pr *v1.PullRequestUnderTest, mimickedJob string, extraOptions ...options) (*prowv1.ProwJob, error) {
```

Comment:
One thing that must change in this method is this part on L468 - the individual aggregated jobs must not share a namespace, each one of them must have an  additional (different) custom job hash input

```
// Avoid sharing when we run the same job multiple times.
// PRPQR name should be safe to use as a discriminating input, because
// there should never be more than one execution of a specific job per
// PRPQR (until aggregated jobs, but for them we'll have a sequence index)
jobBaseGen.PodSpec.Add(prowgen.CustomHashInput(prpqrName))
```